File: p/LocaleDataDemo/app/src/main/java/com/thomaskuenneth/localedatademo/MainActivity.java
package com.thomaskuenneth.localedatademo;

import android.app.Activity;
import android.icu.util.LocaleData;
import android.icu.util.ULocale;
import android.os.Bundle;
import android.util.Log;

import java.util.Locale;

public class MainActivity extends Activity {

    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ULocale locale = ULocale.forLocale(Locale.GERMANY);
        LocaleData ld = LocaleData.getInstance(locale);
        Log.i(TAG, ld.getDelimiter(LocaleData.QUOTATION_START));
        Log.i(TAG, ld.getDelimiter(LocaleData.QUOTATION_END));
        Log.i(TAG, ld.getDelimiter(LocaleData.ALT_QUOTATION_START));
        Log.i(TAG, ld.getDelimiter(LocaleData.ALT_QUOTATION_END));

        LocaleData.PaperSize ps = LocaleData.getPaperSize(locale);
        Log.i(TAG, String.format("width: %d, height: %d",
                ps.getWidth(),
                ps.getHeight()));

        LocaleData.MeasurementSystem ms = LocaleData.getMeasurementSystem(locale);
        if (LocaleData.MeasurementSystem.SI == ms) {
            Log.i(TAG, "SI");
        } else if (LocaleData.MeasurementSystem.UK == ms) {
            Log.i(TAG, "UK");
        } else if (LocaleData.MeasurementSystem.US == ms) {
            Log.i(TAG, "US");
        } else {
            Log.i(TAG, "unknown");
        }
    }
}

File: s/SplashScreenDemo/app/src/main/java/com/thomaskuenneth/splashscreendemo/MainActivity.kt
package com.thomaskuenneth.splashscreendemo

import android.animation.AnimatorSet
import android.animation.ValueAnimator
import android.os.Bundle
import android.util.Log
import android.view.animation.AccelerateDecelerateInterpolator
import android.window.SplashScreen
import androidx.appcompat.app.AppCompatActivity


private const val TAG = "MainActivity"

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "--> ${splashScreen::class.java.name}")
        setContentView(R.layout.activity_main)
        val l = SplashScreen.OnExitAnimationListener { view ->
            Log.d(TAG, view::class.java.name)
            view.iconView?.let { icon ->
                val animator = ValueAnimator
                    .ofInt(icon.height, 0)
                    .setDuration(2000)
                animator.addUpdateListener {
                    val value = it.animatedValue as Int
                    icon.layoutParams.width = value
                    icon.layoutParams.height = value
                    icon.requestLayout()
                    if (value == 0) {
                        view.remove()
                    }
                }
                val animationSet = AnimatorSet()
                animationSet.interpolator = AccelerateDecelerateInterpolator()
                animationSet.play(animator);
                animationSet.start()
            }
        }
        splashScreen.setOnExitAnimationListener(l)
    }
}